sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/higherOrderFunctions.scala

```diff
@@ -357,10 +359,23 @@ case class ArrayFilter(
 
   override def bind(f: (Expression, Seq[(DataType, Boolean)]) => LambdaFunction): ArrayFilter = {
     val ArrayType(elementType, containsNull) = argument.dataType
-    copy(function = f(function, (elementType, containsNull) :: Nil))
+    function match {
+      case LambdaFunction(_, arguments, _) if arguments.size == 2 =>
+        copy(function = f(function, (elementType, containsNull) :: (IntegerType, false) :: Nil))
+      case _ =>
+        copy(function = f(function, (elementType, containsNull) :: Nil))
```

Comment:
We don't need to validate # of arguments here? (the case: arguments.size > 2)

Reply:
Can you check the current error mesasage for the case (arguments.size > 2)?

Reply:
ArrayTransform doesn't validate arguments.size > 2. I'm not sure what happens in that case either.

Reply:
nvm. I checked the error handling works well for the case, too.

Reply:
Yes, it does. See the test here: https://github.com/apache/spark/pull/25666/files#diff-8e1a34391fdefa4a3a0349d7d454d86fR2204.